saleor/site/utils.py

```diff
@@ -0,0 +1,20 @@
+from django.conf import settings
+
+from .models import SiteSetting
+
+
+def get_site_settings(request):
+    if hasattr(request, 'site_settings'):
+        site_settings = request.site_settings
+    else:
+        site_settings_id = getattr(settings, 'SITE_SETTINGS_ID', None)
+        try:
+            site_settings = get_site_settings_uncached(site_settings_id)
+        except SiteSetting.DoesNotExist:
+            site_settings = None
```

Comment:
Do we ever want this to fail? I'm afraid we'll be forced to handle this case all over the place. I see two solutions: either let the exception propagate or call `get_or_create`. Both have their downsides but I thin both are better than failing silently and causing other code to break.